File: spec/factories/recipes.rb
# == Schema Information
#
# Table name: recipes
#
#  id                   :integer          not null, primary key
#  required_item_id1    :integer
#  required_item_count1 :integer
#  required_item_id2    :integer
#  required_item_count2 :integer
#  required_item_id3    :integer
#  required_item_count3 :integer
#  required_item_id4    :integer
#  required_item_count4 :integer
#  required_item_id5    :integer
#  required_item_count5 :integer
#  product_item_id      :integer
#  product_item_count   :integer
#  created_at           :datetime         not null
#  updated_at           :datetime         not null
#

FactoryGirl.define do
  factory :recipe do
    required_item_id1 1
required_item_num1 1
required_item_id2 1
required_item_num2 1
required_item_id3 1
required_item_num3 1
required_item_id4 1
required_item_num4 1
required_item_id5 1
required_item_num5 1
product_item_id 1
product_item_num 1
  end

end

File: packages/db/src/schema/recipes.ts
import { pgTable, serial, integer, timestamp } from "drizzle-orm/pg-core";
import { items } from "./items";

export const recipes = pgTable("recipes", {
  id: serial("id").primaryKey(),
  requiredItemId1: integer("required_item_id1").references(() => items.id),
  requiredItemCount1: integer("required_item_count1"),
  requiredItemId2: integer("required_item_id2").references(() => items.id),
  requiredItemCount2: integer("required_item_count2"),
  requiredItemId3: integer("required_item_id3").references(() => items.id),
  requiredItemCount3: integer("required_item_count3"),
  requiredItemId4: integer("required_item_id4").references(() => items.id),
  requiredItemCount4: integer("required_item_count4"),
  requiredItemId5: integer("required_item_id5").references(() => items.id),
  requiredItemCount5: integer("required_item_count5"),
  productItemId: integer("product_item_id")
    .notNull()
    .references(() => items.id),
  productItemCount: integer("product_item_count").notNull(),
  skillId: integer("skill_id"),
  difficulty: integer("difficulty"),
  createdAt: timestamp("created_at").notNull().defaultNow(),
  updatedAt: timestamp("updated_at").notNull().defaultNow(),
});
